tiledb/api/src/query/read/aggregate.rs

```diff
@@ -0,0 +1,665 @@
+                        ffi::tiledb_channel_operator_max_get(
+                            ctx,
+                            &mut agg_operator,
+                        )
+                    })?;
+                }
+                AggregateType::Min(_) => {
+                    context.capi_call(|ctx| unsafe {
+                        ffi::tiledb_channel_operator_min_get(
+                            ctx,
+                            &mut agg_operator,
+                        )
+                    })?;
+                }
+                AggregateType::Mean(_) => {
+                    context.capi_call(|ctx| unsafe {
+                        ffi::tiledb_channel_operator_mean_get(
+                            ctx,
+                            &mut agg_operator,
+                        )
+                    })?;
+                }
+                AggregateType::Count => unreachable!(),
+            };
+            context.capi_call(|ctx| unsafe {
+                ffi::tiledb_create_unary_aggregate(
+                    ctx,
+                    cquery,
+                    agg_operator,
+                    c_attr_name,
+                    &mut agg_operation,
+                )
+            })?;
+        }
+
+        // Apply aggregate to the default channel.
+        context.capi_call(|ctx| unsafe {
+            ffi::tiledb_channel_apply_aggregate(
+                ctx,
+                default_channel,
+                c_agg_name,
+                agg_operation,
+            )
+        })?;
+
+        Ok(AggregateTypedBuilder::<Self, T> {
+            base: self,
+            agg_str: agg_name,
+            attr_str: attr_name,
+            attr_type: PhantomData,
+        })
+    }
+
+    fn count(self) -> TileDBResult<AggregateTypedBuilder<Self, u64>> {
+        self.apply_typed_aggregate::<u64>(AggregateType::Count)
+    }
+
+    fn nullcount(
+        self,
+        attr_name: String,
+    ) -> TileDBResult<AggregateTypedBuilder<Self, u64>> {
+        self.apply_typed_aggregate::<u64>(AggregateType::NullCount(attr_name))
+    }
+
+    fn mean(
+        self,
+        attr_name: String,
+    ) -> TileDBResult<AggregateTypedBuilder<Self, f64>> {
+        self.apply_typed_aggregate::<f64>(AggregateType::Mean(attr_name))
+    }
+
+    fn sum<T: 'static>(
+        self,
+        attr_name: String,
+    ) -> TileDBResult<AggregateTypedBuilder<Self, T>> {
+        self.apply_typed_aggregate::<T>(AggregateType::Sum(attr_name))
+    }
+
+    fn min<T: 'static>(
+        self,
+        attr_name: String,
+    ) -> TileDBResult<AggregateTypedBuilder<Self, T>> {
+        self.apply_typed_aggregate::<T>(AggregateType::Min(attr_name))
+    }
+
+    fn max<T: 'static>(
+        self,
+        attr_name: String,
+    ) -> TileDBResult<AggregateTypedBuilder<Self, T>> {
+        self.apply_typed_aggregate::<T>(AggregateType::Max(attr_name))
+    }
+}
+
+pub trait AggregateEnumBuilderTrait: QueryBuilder {
```

Comment:
the bounds will be simpler if you just require `AggregateBuilderTrait` for the whole trait.